### ISSUE TITLE
path.resolve could be a problem

### ISSUE THREAD

**Issue by Wyzix33** [kolodziejczak-sz/uwebsocket-serve#4]
I think you could use some customized implementation of path.resolve, which does not escape the parent, I've used something like this
```
function saferesolve(base, target) {
  var targetPath = '.' + path.posix.normalize('/' + target);
  return path.posix.resolve(base, targetPath);
}
```
and even 
```
 saferesolve(base, target) {
  var DANGEROUS_PATH_PATTERN = /(?:^|[\\/])\.\.(?:[\\/]|$)/;
  if (target.indexOf('\0') !== -1 || DANGEROUS_PATH_PATTERN.test(target)) return false;
  return base + path.posix.normalize('/' + target);
 },
```

**Comment by snkj5** [comment]
Hey, thanks for your issue! I think our current code (1.0.5) thanks to @Sup3rFire is preventing path escaping from the parent dir. Take a look here: https://github.com/kolodziejczak-sz/uwebsocket-serve/blob/f088c56d6fb211ebc0b8e5d9fb28e68a377fae99/src/index.ts#L10

Your code looks ok in that matter as well, but can you provide some examples (urls) where your approach might be better?

**Comment by Wyzix33** [comment]
I'm sorry for the late response, i was busy with a new project.. There was a potential path traversal vulnerability found in the 'path.resolve' function. It could occur on a Windows server as it allows the use of backward slash('\') characters as separators in a file path allowing the user to move beyond the root folder defined for that user. But i am unable to reproduce it now...
Also you can add yarn/npm commands for installing this lib `yarn add uwebsocket-serve@kolodziejczak-sz/uwebsocket-serve#v1.0.7` or any release

**Comment by snkj5** [comment]
Hello! No problem, my response took some time as well

> But i am unable to reproduce it now...

I guess you had the v1.0.4 or older version, because like I said other contributor discovered this issue as well and we've managed to push fix for it


> Also you can add yarn/npm commands for installing this lib `yarn add uwebsocket-serve@kolodziejczak-sz/uwebsocket-serve#v1.0.7` or any release

I released the v1.0.6 and v1.0.7 versions via github so you should be able to do so.

Also, don't hasitate to open a new issue in the future, I'll try my best to update the package
